datatable/javascript/src/data_table.js

```diff
@@ -0,0 +1,69 @@
+class DataTable {
+  // TODO: jsdoc
+  // TODO: fail on unequal rows?
+
+  constructor(rawTable) {
+    this._rawTable = rawTable
+  }
+
+  get cells() {
+    return this._rawTable
+  }
+
+  get height() {
+    return this._rawTable.length
+  }
+
+  get width() {
+    return this._rawTable[0].length
+  }
+
+  get isEmpty() {
+    return this.height <= 1 && this.width === 0
+  }
+
+  static create(rawTable) {
```

Comment:
I added this in Java so I could hide the constructor. This makes it easier to add in other `create` methods later on that do some preprocessing before constructing the table. 

You can't do this in JS so I think create is not needed.